packages/headless/src/features/commerce/product-templates/product-templates-manager.ts

```diff
@@ -0,0 +1,40 @@
+import {ProductRecommendation} from '../../../api/search/search/product-recommendation';
+import {
+  buildTemplatesManager,
+  Template,
+  TemplateCondition,
+} from '../../templates/templates-manager';
+
+export type ProductTemplate<Content = unknown> = Template<
+  ProductRecommendation,
+  Content
+>;
+export type ProductTemplateCondition = TemplateCondition<ProductRecommendation>;
+
+export interface ProductTemplatesManager<Content = unknown> {
+  /**
+   * Registers any number of result templates in the manager.
+   * @param templates (...ResultTemplate<Content>) A list of templates to register.
+   */
+  registerTemplates(
+    ...newTemplates: Template<ProductRecommendation, Content>[]
+  ): void;
+  /**
+   * Selects the highest priority template for which the given result satisfies all conditions.
+   * In the case where satisfied templates have equal priority, the template that was registered first is returned.
+   * @param result (Result) The result for which to select a template.
```

Comment:
```suggestion
   * @param product (Product) The product for which to select a template.
```

(once again, preemptive documented type name suggestion).